weblate/api/views.py

```diff
@@ -896,6 +897,28 @@ def units(self, request, **kwargs):
 
         return self.get_paginated_response(serializer.data)
 
+    @action(detail=True, methods=["post"])
+    def add_monolingual(self, request, **kwargs):
+        obj = self.get_object()
```

Comment:
Should also check for permissions here: request.user.has_perm("unit.add", translation)